dvc/scm/git/backend/pygit2.py

```diff
@@ -93,7 +93,25 @@ def commit(self, msg: str, no_verify: bool = False):
     def checkout(
         self, branch: str, create_new: Optional[bool] = False, **kwargs,
     ):
-        raise NotImplementedError
+        from pygit2 import GitError
+
+        if create_new:
+            commit = self.repo.revparse_single("HEAD")
+            new_branch = self.repo.branches.local.create(branch, commit)
+            self.repo.checkout(new_branch)
+        else:
+            if branch == "-":
+                branch = "@{-1}"
```

Comment:
For whatever reason, libgit2's `resolve_refish` doesn't handle `-` shorthand, probably because it's specific to `git checkout/switch` and not a formal/general git ref shorthand symbol